### PR TITLE
pre-commit: Optimize SVGs automatically

### PR DESCRIPTION
**Pull request by Holzhaus**
It get annoying to remind people to optimize SVGs using Inkscape... and apparently that isn't even an option on WIndows.

### CODE REVIEW

File: tools/optimize_svg.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+ARGS = (
+    "--enable-comment-stripping",
+    "--enable-id-stripping",
+    "--enable-viewboxing",
+    "--indent=none",
+    "--no-line-breaks",
+    "--remove-descriptions",
+    "--remove-metadata",
+    "--remove-titles",
```
Comment (ronso0) [inline tools/optimize_svg.py:15]: all done by --remove-descriptive-elements:
```
    --remove-descriptive-elements
                                 remove <title>, <desc> and <metadata>
                                 elements

```

**Review by Swiftb0y — APPROVED** [review]

**Comment by Swiftb0y** [comment]
@Holzhaus @ronso0 do you think it would make sense to add the same pre-commit hook to the main repo for minimizing the graphics used in skins?

**Comment by Holzhaus** [comment]
We can. Maybe I'll make a separate repo so that we can avoid duplicating the code in each repo.

PS: I wish scour supported conversion into SVG Tiny. That would solve a lot of headaches.

**Comment by ronso0** [comment]
I agree with @esbrandt 
In existing skins most graphics are optimized already, and the maintainer takes care of new icons.
For new skins we need to request it only once (not repeatedly like for mapping documentations).

Also, with the Qt skins the view in the vector program is not necessarily WYSIWYG as soon as gradients and transparencies are used. then we cannot rely on the script unfortunately, icons needs to be checked manually.

**Pull request merged**